packages/babel-plugin-transform-destructuring/test/fixtures/assumption-iterableIsArray/for-in/output.js

```diff
@@ -5,7 +5,8 @@ for (var _ref in obj) {
 }
 
 for (var _ref2 in obj) {
-  name = _ref2[0];
-  value = _ref2[1];
+  var _ref3 = _ref2;
+  name = _ref3[0];
+  value = _ref3[1];
```

Comment:
Input: 
```js
for ([name, value] in obj) {
```

This output changes because the destructuring transform first transforms it to
```js
for (var _ref2 in obj) {
  [name, value] = _ref2;
```
and then `_ref2` isn't considered constant anymore.

We can avoid this in Babel 8, by injecting `let` instead of `var`.